django_cloud_deploy/workflow/_project.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manages Google Cloud Platform projects creation."""
+
+import enum
+
+from django_cloud_deploy.cloudlib import project
+
+from google.auth import credentials
+
+
+class CreationMode(enum.Enum):
```

Comment:
Why do we need CreationMode? Don't we always want to create it ? Is this for tests to not create an already exisiting project? Not sure why MUST_EXIST is needed.

Reply:
Because we cannot create new projects, this is useful for our integration test.

Reply:
It also might be useful if we let the user add a Django cluster to an existing project.

Reply:
So we would want to ask the user if they would like to create a new project or use an existing one?

If they use an exisiting one they have to choose from a list or pass in the ID?

And for a new one could follow the exisiting flow.